From ff3152a23dde368ef0bfd51aafb88f00e4c015f5 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 07 Jun 2022 14:33:39 +0300
Subject: [PATCH] [SW-0] gaudi3: perform pcie config based on security enable bit

As part of ROM recovery, it is decided to open complete config space and
bring up pcie link mainly for debug purpose. This option wont be
available on production system. This patch perform PCIe configuration
based on SEC_ENABLE efuse bit.

Change-Id: I8f1fffc1261433cb668542c2e109f5811d2a41f5
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 8525c3a..be55887 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -166,7 +166,9 @@
 	if (!arcboot_rom_global.die_id)
 		security_rr_recovery_config();
 
-	rom_pcie_recovery();
+	if (!(arcboot_rom_global.config_fuse & EFUSE_SYSCFG_SECURITY_ENABLE_MASK))
+		rom_pcie_recovery();
+
 	rom_d2d_link_init();
 }
 
